### PR TITLE
Add per-Task cpu time metric (#56320)

### PR DESCRIPTION
**Pull request by nickrobinson251**
## PR Description

Backports https://github.com/JuliaLang/julia/pull/56320.

## Checklist

Requirements for merging:
- [x] I have opened an issue or PR upstream on JuliaLang/julia: https://github.com/JuliaLang/julia/pull/56320
- [x] I have removed the `port-to-*` labels that don't apply.
- [X] I have opened a PR on raicode to test these changes: https://github.com/RelationalAI/raicode/pull/22280

### CODE REVIEW

File: doc/man/julia.1 [hunk 1]
```diff
@@ -270,6 +270,14 @@ Print precompile statements for methods compiled during execution or save to a p
 --trace-compile-timing=
 If --trace-compile is enabled show how long each took to compile in ms
 
+.TP
+--trace-dispatch={stderr|name}
+Print precompile statements for methods dispatched during execution or save to stderr or a path.
```
Comment (NHDaly) [inline doc/man/julia.1:275]: This is also a compiler cache change, right? Can we separate it?

File: base/experimental.jl [hunk 2]
```diff
@@ -368,4 +368,87 @@ adding them to the global method table.
...
 :@MethodTable
 
+"""
+    Base.Experimental.disable_new_worlds()
+
+Mark that no new worlds (methods additions, deletions, etc) are permitted to be created at
+any future time, allowing for lower latencies for some operations and slightly lower memory
+usage, by eliminating the tracking of those possible invalidation.
+"""
+disable_new_worlds() = ccall(:jl_disable_new_worlds, Cvoid, ())
```
[the remark this answers is not in the document]
Comment (kpamnany) [reply base/experimental.jl:378]: We already backported the C `jl_disable_new_worlds` so I thought to leave this in when the cherry pick brought it in. But you're right that it isn't really part of this backport. Should I just drop it?

**Review by NHDaly — APPROVED** [review]
LGTM! Glad to finally have this! :)

File: base/experimental.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (kpamnany) [reply base/experimental.jl]: Dropping this.

**Pull request merged**